diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BracesInConditionalsAndLoopsRule.kt

```diff
@@ -90,6 +92,13 @@ class BracesInConditionalsAndLoopsRule(configRules: List<RulesConfig>) : DiktatR
         }
 
         if (elseKeyword != null && elseNode?.elementType != IF && elseNode?.elementType != BLOCK) {
+            val referenceExpressionChildren = elseNode?.findAllDescendantsWithSpecificType(REFERENCE_EXPRESSION)
```

Comment:
Shouldn't we check only top-level children? IIRC, `findAllDescendantsWithSpecificType` will return even deeply nested nodes.

Reply:
Yeah, they reason, why it can be dangerous, that in theory, we can take some wrong node, from the nested ones
Initially I can't imagine some proper case, and thought that it will worked anyway, however now I did this, add more tests and little bit changed the logic, now we looking into the direct children